lib/Db/PersonMapper.php

```diff
@@ -193,6 +193,14 @@ public function mergeClusterToDatabase(string $userId, $currentClusters, $newClu
 
 				$oldFaces = $currentClusters[$newPerson];
 				if ($newFaces === $oldFaces) {
+					// Set cluster as valid now
```

Comment:
you might not believe me, but I am laughing now:) I noticed this "continue" when I was working on a fix and I thought "nah, no way this can even happen"...and guess what - it happened:)

Can you try adding:
```php
$personMapper->invalidatePersons($image->getId());
```

in:
* https://github.com/matiasdelellis/facerecognition/blob/master/tests/Integration/MergeClusterToDatabaseTest.php#L73

I think this case should be covered now?

Reply:
hahah.. Ok ok.. :sweat_smile: 
Sorry.. I will leave the tests to you.. :wink: 
Although I am aware that the last changes we should add for each one a test ....